src/account/account.service.ts

```diff
@@ -37,8 +46,44 @@ export class AccountService {
     constructor(
         private readonly db: Knex,
         private readonly events: EventEmitter,
+        private readonly accountRepository: KnexAccountRepository,
+        private readonly fedifyContextFactory: FedifyContextFactory,
     ) {}
 
+    /**
+     * Get an Account by the ActivityPub ID
+     * If it is not found locally in our database it will be
+     * remotely fetched and stored
+     */
+    async getByApId(id: URL): Promise<Account> {
+        const account = await this.accountRepository.getByApId(id);
+        if (account) {
+            return account;
+        }
+
+        const context = this.fedifyContextFactory.getFedifyContext();
+
+        const documentLoader = await context.getDocumentLoader({
+            handle: 'index',
+        });
+        const potentialActor = await lookupObject(id, { documentLoader });
```

Comment:
🤔 We have `FedifyActorResolver` thats used by the publishing service. Could we utilise the same thing here?

Reply:
Yeah so this is annoying - the `lookupActor` method which the `FedifyActorResolver` uses is only available on a `FedifyRequestContext` - but we only have access to a `Fedify` context because the inbox handlers aren't run in the context of a request - they're run in the context of the queue